src/calculate_bucket.hpp

```diff
@@ -77,7 +77,9 @@ class F1Calculator {
     inline F1Calculator(uint8_t k, const uint8_t* orig_key)
     {
         uint8_t enc_key[32];
+        size_t buf_blocks = cdiv((k + kExtraBits) << kBatchSizes, kF1BlockSizeBits) + 2;
         this->k_ = k;
+        this->buf_ = (uint8_t *)malloc(buf_blocks * kF1BlockSizeBits / 8);
```

Comment:
if you use `new uint8_t[buf_blocks * kF1BlockSizeBits / 8]` instead, you don't need to check for null. But then you'd need to replace `free(this->buf_);` with `delete[] this->buf_;`

Although, ideally, `buf_` would be of type `std::unique_ptr<uint8_t[]>` and you wouldn't call `free()` or `delete[]` at all. I intend to propose a change like that in other places too.